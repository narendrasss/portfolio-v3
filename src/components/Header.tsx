import React from 'react';
import { Flex } from 'rebass';
import styled from 'styled-components';
import Domain from './Domain';
import List from './base/List';
import Link from './base/Link';
import Box from './base/Box';

export interface HeaderProps {
  url: string;
  links: LinkTo[];
}

const Container = styled(Flex)`
  position: fixed;
  top: 0;
  left: 0;
`;

const Header: React.FC<HeaderProps> = ({ url, links }) => {
  return (
    <Container as="header" width="100%" m={1} justifyContent="space-between">
      <Domain url={url} />
      <Box display={['none', 'block']}>
        <List items={links}>
          {({ name, url }) => (
            <Link key={[name, url].join()} to={url}>
              {name}
            </Link>
          )}
        </List>
      </Box>
    </Container>
  );
};

export default Header;
