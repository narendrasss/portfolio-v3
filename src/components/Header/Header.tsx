import React from 'react';
import { Flex } from 'rebass';
import styled from 'styled-components';
import Domain from '../Domain/Domain';
import List from '../base/List';
import Link from '../base/Link';
import Box from '../base/Box';

export interface HeaderProps {
  url: string;
  links: LinkTo[];
}

const Container = styled(Flex)`
  position: fixed;
  top: 0;
  left: 0;
  justify-content: space-between;
  width: 100%;
`;

const Header: React.FC<HeaderProps> = ({ url, links }) => {
  return (
    <Container as="header" m={1}>
      <Domain url={url} />
      <Box as="nav" display={['none', 'block']}>
        <List items={links}>
          {({ name, url }) => (
            <Link key={[name, url].join()} to={url}>
              {name}
            </Link>
          )}
        </List>
      </Box>
    </Container>
  );
};

export default Header;
