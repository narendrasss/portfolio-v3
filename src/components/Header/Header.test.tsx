import React from 'react';
import { render } from 'react-testing-library';
import Header from './Header';

describe('Header', () => {
  const url = 'narendras.io';
  const links = [
    {
      name: 'about',
      url: '/about',
    },
    {
      name: 'blog',
      url: '/blog',
    },
  ];
  const component = <Header url={url} links={links} />;

  it('renders without crashing', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all links correctly', () => {
    const { queryByText } = render(component);
    links.forEach(({ name }) => {
      expect(queryByText(name)).toBeTruthy();
    });
  });
});
