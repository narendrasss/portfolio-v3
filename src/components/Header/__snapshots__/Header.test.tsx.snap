// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<DocumentFragment>
  <header
    class="sc-bdVaJa sc-bwzfXH sc-iwsKbI dUmjld"
    width="100%"
  >
    <a
      class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat cYLLdT"
      color="black"
      font-family="sans"
      font-size="0,1"
      font-weight="600"
      href="/home"
    >
      narendras
      <span
        class="sc-bdVaJa sc-htpNat fYauba"
        font-family="sans"
        font-size="0,1"
        font-weight="300"
      >
        .io
      </span>
    </a>
    <nav
      class="sc-bdVaJa sc-dnqmqq hjWSCW"
      display="none,block"
    >
      <a
        class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
        color="black"
        font-family="sans"
        font-size="0,1"
        href="/about"
      >
        about
      </a>
      <a
        class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
        color="black"
        font-family="sans"
        font-size="0,1"
        href="/blog"
      >
        blog
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
